rules_daml/daml.bzl

```diff
@@ -1,94 +1,283 @@
 # Copyright (c) 2020 The DAML Authors. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-load("@bazel_skylib//lib:paths.bzl", "paths")
 load("@build_environment//:configuration.bzl", "ghc_version")
 
-daml_provider = provider(doc = "DAML provider", fields = {
-    "dalf": "The DAML-LF file.",
-    "dar": "The packaged archive.",
-})
-
-def _daml_impl_compile_dalf(ctx):
-    # Call damlc compile
-    compile_args = ctx.actions.args()
-    compile_args.add("compile")
-    compile_args.add(ctx.file.main_src)
-    compile_args.add("--output", ctx.outputs.dalf.path)
-    if ctx.attr.target:
-        compile_args.add("--target", ctx.attr.target)
-    ctx.actions.run(
-        inputs = depset([ctx.file.main_src] + ctx.files.srcs),
-        outputs = [ctx.outputs.dalf],
-        arguments = [compile_args],
-        progress_message = "Compiling DAML into DAML-LF archive %s" % ctx.outputs.dalf.short_path,
-        executable = ctx.executable.damlc,
-    )
-
-def _daml_impl_package_dar(ctx):
-    # Call damlc package
-    package_args = ctx.actions.args()
-    package_args.add("package")
-    package_args.add(ctx.file.main_src)
-    package_args.add(ctx.attr.name)
-    if ctx.attr.target:
-        package_args.add("--target", ctx.attr.target)
-    package_args.add("--output")
-    package_args.add(ctx.outputs.dar.path)
-    ctx.actions.run(
-        inputs = [ctx.file.main_src] + ctx.files.srcs,
-        outputs = [ctx.outputs.dar],
-        arguments = [package_args],
-        progress_message = "Creating DAR package %s" % ctx.outputs.dar.basename,
-        executable = ctx.executable.damlc,
-    )
-
-def _daml_compile_impl(ctx):
-    _daml_impl_compile_dalf(ctx)
-    _daml_impl_package_dar(ctx)
-
-    # DAML provider
-    daml = daml_provider(
-        dalf = ctx.outputs.dalf,
-        dar = ctx.outputs.dar,
-    )
-    return [daml]
-
-def _daml_compile_outputs_impl(name):
-    patterns = {
-        "dalf": "{name}.dalf",
-        "dar": "{name}.dar",
-    }
-    return {
-        k: v.format(name = name)
-        for (k, v) in patterns.items()
-    }
-
-daml_compile = rule(
-    implementation = _daml_compile_impl,
+_damlc = attr.label(
+    allow_single_file = True,
+    default = Label("//compiler/damlc"),
+    executable = True,
+    cfg = "host",
+    doc = "The DAML compiler.",
+)
+
+def _daml_configure_impl(ctx):
+    project_name = ctx.attr.project_name
+    project_version = ctx.attr.project_version
+    daml_yaml = ctx.outputs.daml_yaml
+    target = ctx.attr.target
+    ctx.actions.write(
+        output = daml_yaml,
+        content = """
+            sdk-version: 0.0.0
```

Comment:
Please use the actual SDK version here instead of `0.0.0`. This doesn’t matter too much for DARs used for tests but it does matter for DARs like the trigger library since `daml build` refuses to use `dependencies` with a different `sdk-version` field (for good reasons).

Given that we did actually just fix this for `damlc package` and therefore `daml_compile` rules in #4999, this would be a regression so let’s please fix this in this PR.

Reply:
How does it make any different what it is set to?
We are using `damlc` directly, and not going via the SDK/assistant.

Reply:
The `sdk-version` that you set here is what ends up in the `MANIFEST.MF` in the DAR.

Reply:
ok, I fixed this now